lib/gems/pending/util/miq-password.rb

```diff
@@ -23,21 +23,23 @@ def encrypt(str, ver = "v2", key = self.class.keys[ver])
     "#{ver}:{#{value}}"
   end
 
-  def decrypt(str, legacy = false)
-    if str.nil? || str.empty?
-      str
-    else
-      ver, enc = self.class.split(str)
-      return "" if enc.empty?
+  def decrypt(str, legacy = false, key = nil)
```

Comment:
I'm not sure if it's worth doing now/in this PR, but I have no idea what `legacy` means in this context. Can we rename it to something better?